modules/ppcp-api-client/src/Helper/PurchaseUnitSanitizer.php

```diff
@@ -0,0 +1,349 @@
+<?php
+/**
+ * Class PurchaseUnitSanitizer.
+ *
+ * Sanitizes a purchase_unit array to be consumed by PayPal.
+ *
+ * All money values send to PayPal can only have 2 decimal points. WooCommerce internally does
+ * not have this restriction. Therefore, the totals of the cart in WooCommerce and the totals
+ * of the rounded money values of the items, we send to PayPal, can differ. In those case we either:
+ * - Add an extra line with roundings.
+ * - Don't send the line items.
+ *
+ * @package WooCommerce\PayPalCommerce\ApiClient\Helper
+ */
+
+declare(strict_types=1);
+
+namespace WooCommerce\PayPalCommerce\ApiClient\Helper;
+
+use WooCommerce\PayPalCommerce\ApiClient\Entity\Item;
+use WooCommerce\PayPalCommerce\ApiClient\Entity\Money;
+
+/**
+ * Class PurchaseUnitSanitizer
+ */
+class PurchaseUnitSanitizer {
+	const MODE_DITCH      = 'ditch';
+	const MODE_EXTRA_LINE = 'extra_line';
+	const VALID_MODES     = array(
+		self::MODE_DITCH,
+		self::MODE_EXTRA_LINE,
+	);
+
+	const EXTRA_LINE_NAME = 'Subtotal mismatch';
+
+	/**
+	 * The purchase unit data
+	 *
+	 * @var array
+	 */
+	private $purchase_unit = array();
+
+	/**
+	 * Whether to allow items to be ditched.
+	 *
+	 * @var bool
+	 */
+	private $allow_ditch_items = true;
+
+	/**
+	 * The working mode
+	 *
+	 * @var string
+	 */
+	private $mode;
+
+	/**
+	 * The name for the extra line
+	 *
+	 * @var string
+	 */
+	private $extra_line_name;
+
+	/**
+	 * The last message. To be added to order notes.
+	 *
+	 * @var string
+	 */
+	private $last_message = '';
+
+	/**
+	 * PurchaseUnitSanitizer constructor.
+	 *
+	 * @param string|null $mode The mismatch handling mode, ditch or extra_line.
+	 * @param string|null $extra_line_name The name of the extra line.
+	 */
+	public function __construct( string $mode = null, string $extra_line_name = null ) {
+
+		if ( ! in_array( $mode, self::VALID_MODES, true ) ) {
+			$mode = self::MODE_DITCH;
+		}
+
+		if ( ! $extra_line_name ) {
+			$extra_line_name = self::EXTRA_LINE_NAME;
+		}
+
+		$this->mode            = $mode;
+		$this->extra_line_name = $extra_line_name;
+	}
+
+	/**
+	 * Indicates if mode is ditch.
+	 *
+	 * @return bool
+	 */
+	private function is_mode_ditch(): bool {
+		return $this->mode === self::MODE_DITCH;
```

Comment:
I would just check the `mode` without creating these functions for each possible value.

Also `is_..._mode` would sound better.

Reply:
These functions are useful if there is any logic besides the constant itself, for instance if we had a filter, so no need to do refactors in the future. Also using mode before is visually more organized with autocompletes. But I can change this.